### PR TITLE
fix: harden thread and session lifecycle ownership

### PR DESCRIPTION
**Pull request by minpeter**
## Summary

- retain the authoritative cached thread handle across lifecycle failures
- fence stale handles with identity-aware eviction and coordinated delete/dispose operations
- keep the legacy `new-session` action source-compatible as a deprecated no-op, removing its destructive thread-deletion path
- add regressions for failed deletion, stale disposal/deletion, lifecycle races, Agent disposal failures, and legacy command payloads

## Review fixes

- repeated calls on a stale disposed handle cannot evict a replacement handle
- stale disposed handles cannot delete durable state owned by a replacement
- concurrent delete/dispose follows one lifecycle operation; failed deletion remains retry-authoritative
- `Agent.dispose()` retains handles whose disposal failed
- extension migration guidance is documented in the README and Tegami note

## Verification

- `bun run --cwd packages/runtime test` — 1,704 passed, 5 skipped
- `bun run --cwd apps/coding-agent test` — 884 passed
- runtime and coding-agent typechecks and lints
- runtime and coding-agent builds
- `bun run boundaries`
- `bun run api:check`
- release artifact verification
- manual stale-handle security driver: replacement remains cached, stale delete rejects, durable state remains
- built TUI QA with a legacy extension action; durable FileHost state remained intact

## Release

- patch Tegami entries for `@minpeter/pss-runtime` and `@minpeter/pss-coding-agent`
- deprecated `new-session` action migration behavior documented

### CODE REVIEW

File: apps/coding-agent/src/tui/app.ts [hunk 1]
```diff
@@ -536,37 +536,21 @@ export async function startTui(
...
+        try {
+          await reloadExtensionRuntime();
+        } catch (error) {
+          // A failed reload may have touched the durable thread. Successful
+          // disposal evicts the cached handle so this request re-reads the
+          // store; failed disposal leaves the retryable handle authoritative.
+          const stale = thread;
+          stale.interrupt();
+          await stale.dispose().catch(() => undefined);
+          thread = agent.thread(currentSession.key);
```
[the remark this answers is not in the document]
Comment (minpeter) [reply apps/coding-agent/src/tui/app.ts:561]: Addressed in 43b782d: reload recovery now rebinds only after disposal succeeds; disposal failure surfaces as an AggregateError preserving both causes.

File: apps/coding-agent/src/tui/app-command-action.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    const previousThreadDirectory = process.env.PSS_THREAD_DIR;
+    const previousThreadKey = process.env.PSS_THREAD_KEY;
+    process.chdir(directory);
+    process.env.HOME = directory;
+    process.env.PSS_THREAD_DIR = join(directory, "threads");
+    process.env.PSS_THREAD_KEY = "app-command-action-test";
+    const store = createFileHost({
+      directory: process.env.PSS_THREAD_DIR,
+    }).store.threads;
+    await store.commit(
```
[the remark this answers is not in the document]
Comment (minpeter) [reply apps/coding-agent/src/tui/app-command-action.test.ts]: Addressed in 43b782d: cwd/env mutation, FileHost construction, and seed commit now all execute inside the existing try/finally cleanup boundary.

File: packages/runtime/src/agent/core/agent-dispose.test.ts [hunk 3]
```diff
@@ -26,6 +26,8 @@ describe("Agent.dispose", () => {
 
       expect(outcome).toEqual({ rejection: reason, resolved: false });
       expect(laterDispose).toHaveBeenCalledOnce();
+      expect(agent.thread("first-rejecting-disposal")).toBe(first);
```
[the remark this answers is not in the document]
Comment (minpeter) [reply packages/runtime/src/agent/core/agent-dispose.test.ts:29]: This test intentionally isolates the Agent-level contract: after a public handle disposal rejects, Agent.dispose() must not unconditionally clear that cached entry. The factory-level open/terminal decision is exercised separately in thread-handle-factory.test.ts: "retains an open handle after disposal rejects" and "evicts a terminal handle after disposal rejects" both run the real createThreadPublicHandle lifecycle. Together the layers cover both responsibilities; making this test enter the factory again would duplicate that lower-level coverage and obscure the specific regression (#threads.clear()). The suggested "non-open retention" is not desired: terminal failed disposals must evict, and the dedicated factory test asserts exactly that.
